File: microstartercli/src/main/resources/micronaut/security/api key/java/JDBC/repository/ApiKeyRepository.java
package ${securityPackage}.repository;

import ${securityPackage}.domains.APIKey;
import io.micronaut.data.repository.CrudRepository;
<% if(jdbc) out.print 'import io.micronaut.data.jdbc.annotation.JdbcRepository;'%><% if(mongo) out.print 'import io.micronaut.data.mongodb.annotation.MongoRepository;'%>
<% if(jdbc) out.print 'import io.micronaut.data.model.query.builder.sql.Dialect;'%>

import java.util.Optional;


<% if(jdbc) out.print '@JdbcRepository(dialect = Dialect.'+dialect+')'%><% if(mongo) out.print '@MongoRepository'%>
public interface ApiKeyRepository extends CrudRepository<APIKey, <% if(jdbc) out.print 'Long'%><% if(mongo) out.print 'String'%> > {

    public Optional<APIKey> findByName(String name);
    public Optional<APIKey> findBySecret(String secret);

}

File: MicroCli/src/main/resources/micronaut/security/api key/kotlin/JDBC/repository/ApiKeyRepository.kt
package {securityPackage}.repository


import ${securityPackage}.domains.APIKey
import io.micronaut.data.repository.CrudRepository
import java.util.*


import ${securityPackage}.domains.APIKey
import io.micronaut.data.repository.CrudRepository
<% if(jdbc) out.print 'import io.micronaut.data.jdbc.annotation.JdbcRepository'%><% if(mongo) out.print 'import io.micronaut.data.mongodb.annotation.MongoRepository'%>
<% if(jdbc) out.print 'import io.micronaut.data.model.query.builder.sql.Dialect'%>

import java.util.Optional


<% if(jdbc) out.print '@JdbcRepository(dialect = Dialect.'+dialect+')'%><% if(mongo) out.print '@MongoRepository'%>
interface ApiKeyRepository : CrudRepository<APIKey?, Long?> {
    fun findByName(name: String?): Optional<APIKey?>?
    fun findByKey(key: String?): Optional<APIKey?>?
}


